jsonnet/cmd.go

```diff
@@ -45,6 +168,322 @@ func getVar(s string) (string, string, error) {
+				if err != nil {
+					return false, err
+				}
+				vm.ExtVar(name, content)
+			} else if arg == "--ext-str-file" {
+				next := nextArg(&i, args)
+				name, content, err := getVarFile(next)
+				if err != nil {
+					return false, err
+				}
+				vm.ExtVar(name, content)
+			} else if arg == "--ext-code" {
+				next := nextArg(&i, args)
+				name, content, err := getVarVal(next)
+				if err != nil {
+					return false, err
+				}
+				vm.ExtCode(name, content)
+			} else if arg == "--ext-code-file" {
+				next := nextArg(&i, args)
+				name, content, err := getVarFile(next)
+				if err != nil {
+					return false, err
+				}
+				vm.ExtCode(name, content)
+			} else if arg == "-A" || arg == "--tla-str" {
+				next := nextArg(&i, args)
+				name, content, err := getVarVal(next)
+				if err != nil {
+					return false, err
+				}
+				vm.TLAVar(name, content)
+			} else if arg == "--tla-str-file" {
+				next := nextArg(&i, args)
+				name, content, err := getVarFile(next)
+				if err != nil {
+					return false, err
+				}
+				vm.TLAVar(name, content)
+			} else if arg == "--tla-code" {
+				next := nextArg(&i, args)
+				name, content, err := getVarVal(next)
+				if err != nil {
+					return false, err
+				}
+				vm.TLACode(name, content)
+			} else if arg == "--tla-code-file" {
+				next := nextArg(&i, args)
+				name, content, err := getVarFile(next)
+				if err != nil {
+					return false, err
+				}
+				vm.TLACode(name, content)
+			} else if arg == "-t" || arg == "--max-trace" {
+				l := safeStrToInt(nextArg(&i, args))
+				if l < 0 {
+					usage(os.Stderr)
+					return false, fmt.Errorf("ERROR: Invalid --max-trace value: %d", l)
+				}
+				vm.MaxTrace = l
+			} else if arg == "-m" || arg == "--multi" {
+				config.evalMulti = true
+				outputDir := nextArg(&i, args)
+				if len(outputDir) == 0 {
+					return false, fmt.Errorf("ERROR: -m argument was empty string")
+				}
+				if outputDir[len(outputDir)-1] != '/' {
+					outputDir += "/"
+				}
+				config.evalMultiOutputDir = outputDir
+			} else if arg == "-y" || arg == "--yaml-stream" {
+				config.evalStream = true
+			} else if arg == "-S" || arg == "--string" {
+				vm.StringOutput = true
+			} else if len(arg) > 1 && arg[0] == '-' {
+				return false, fmt.Errorf("ERROR: Unrecognized argument: %s", arg)
+			} else {
+				remainingArgs = append(remainingArgs, arg)
+			}
+
+		} else {
+			return false, fmt.Errorf("The Go implementation currently does not support jsonnet fmt.")
+		}
+	}
+
+	want := "filename"
+	if config.filenameIsCode {
+		want = "code"
+	}
+	if len(remainingArgs) == 0 {
+		usage(os.Stderr)
+		return false, fmt.Errorf("ERROR: Must give %s", want)
+	}
+
+	// TODO(dcunnin): Formatter allows multiple files in test and in-place mode.
+	multipleFilesAllowed := false
+
+	if !multipleFilesAllowed {
+		filename := remainingArgs[0]
+		if len(remainingArgs) > 1 {
+			usage(os.Stderr)
+			return false, fmt.Errorf("ERROR: Already specified %s as \"%s", want, filename)
```

Comment:
The error feels cryptic when all you're disallowing is multiple files to be specified.